localstack/services/sns/sns_listener.py

```diff
@@ -456,13 +467,14 @@ def get_message_attributes(req_data):
     while True:
         name = req_data.get('MessageAttributes.entry.' + str(x) + '.Name', [None])[0]
         if name is not None:
-            attribute = {}
-            attribute['Type'] = req_data.get('MessageAttributes.entry.' + str(x) + '.Value.DataType', [None])[0]
+            attribute = {
+                'Type': req_data.get('MessageAttributes.entry.' + str(x) + '.Value.DataType', [None])[0]
+            }
             string_value = req_data.get('MessageAttributes.entry.' + str(x) + '.Value.StringValue', [None])[0]
             binary_value = req_data.get('MessageAttributes.entry.' + str(x) + '.Value.BinaryValue', [None])[0]
-            if string_value is not None:
+            if string_value:
```

Comment:
Not sure about this change - the updated code would probably behave incorrectly if `string_value` is an empty string (`''`)?